homeassistant/components/weather/zamg.py

```diff
@@ -0,0 +1,106 @@
+"""
+Sensor for data from Austrian "Zentralanstalt für Meteorologie und Geodynamik".
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/weather.zamg/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.weather import (
+    WeatherEntity, ATTR_WEATHER_HUMIDITY, ATTR_WEATHER_PRESSURE,
+    ATTR_WEATHER_TEMPERATURE, ATTR_WEATHER_WIND_BEARING,
+    ATTR_WEATHER_WIND_SPEED, PLATFORM_SCHEMA)
+from homeassistant.const import \
+    CONF_NAME, TEMP_CELSIUS, CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.helpers import config_validation as cv
+# Reuse data and API logic from the sensor implementation
+from homeassistant.components.sensor.zamg import (
+    ATTRIBUTION, closest_station, CONF_STATION_ID, zamg_stations, ZamgData)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_STATION_ID): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the ZAMG sensor platform."""
+    station_id = config.get(CONF_STATION_ID) or closest_station(
+        config.get(CONF_LATITUDE),
+        config.get(CONF_LONGITUDE),
+        hass.config.config_dir)
+    if station_id not in zamg_stations(hass.config.config_dir):
+        _LOGGER.error("Configured ZAMG %s (%s) is not a known station",
+                      CONF_STATION_ID, station_id)
+        return False
+
+    probe = ZamgData(station_id=station_id, logger=_LOGGER)
+    try:
+        probe.update()
+    except ValueError as err:
+        _LOGGER.error("Received error from ZAMG: %s", err)
+        return False
+
+    add_devices([ZamgWeather(probe, config.get(CONF_NAME))], True)
+
+
+class ZamgWeather(WeatherEntity):
+    """Representation of a weather condition."""
+
+    def __init__(self, zamg_data, stationname=None):
+        """Initialise the platform with a data instance and station name."""
+        self.zamg_data = zamg_data
+        self.stationname = stationname or self.zamg_data.data.get('Name')
+
+    def update(self):
+        """Update current conditions."""
+        self.zamg_data.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return 'ZAMG {}'.format(self.stationname or '(unknown station)')
```

Comment:
Sorry, I guess that my thoughts in the first comments were more going in the direction not to hard-code the prefix.